gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaExtractor.java

```diff
@@ -71,7 +75,15 @@ public KafkaExtractor(WorkUnitState state) {
   }
 
   @Override
-  public D readRecord(D reuse) throws DataRecordException, IOException {
+  public List<Tag<?>> generateTags(State state) {
+    List<Tag<?>> tags = super.generateTags(state);
+    tags.add(new Tag<String>("topic", state.getProp(KafkaSource.TOPIC_NAME)));
+    tags.add(new Tag<Integer>("partition", state.getPropAsInt(KafkaSource.PARTITION_ID)));
```

Comment:
`KafkaExtractor` has been modified in #135 to pull from multiple partitions of the same topic. I suggest waiting until that PR is merged to make the corresponding changes here.